kotlin/find-route/src/main/java/com/esri/arcgisruntime/sample/findroute/MainActivity.kt

```diff
@@ -0,0 +1,321 @@
+
+          // get directions
+          // NOTE: to get turn-by-turn directions the route parameters
+          //  must have the isReturnDirections parameter set to true.
+          val directions = route.directionManeuvers
+          val directionsArray = Array<String>(directions.size) { i ->
+            directions[i].directionText
+          }
+
+          if (progressDialog.isShowing) {
+            progressDialog.dismiss()
+          }
+
+          // expand or collapse the bottom sheet when the header is clicked
+          bottomSheet.header.setOnClickListener {
+            bottomSheetBehavior.state = when (bottomSheetBehavior.state) {
+              BottomSheetBehavior.STATE_COLLAPSED -> BottomSheetBehavior.STATE_EXPANDED
+              else -> BottomSheetBehavior.STATE_COLLAPSED
+            }
+          }
+
+          bottomSheet.directionsListView.apply {
+            // Set the adapter for the list view
+            adapter = ArrayAdapter(
+              this@MainActivity,
+              android.R.layout.simple_list_item_1,
+              directionsArray
+            )
+
+            // when the user taps a maneuver, set the viewpoint to that portion of the route
+            onItemClickListener =
+              AdapterView.OnItemClickListener { _, _, position, _ ->
+                // remove any graphics that are not the two stops and the route graphic
+                if (graphicsOverlay.graphics.size > 3) {
+                  graphicsOverlay.graphics.removeAt(graphicsOverlay.graphics.size - 1)
+                }
+
+                // set the viewpoint to the selected maneuver
+                val geometry = directions[position].geometry
+                mapView.setViewpointAsync(
+                  Viewpoint(geometry.extent, 20.0),
+                  1f
+                )
+
+                // create a graphic with a symbol for the maneuver and add it to the graphics overlay
+                val selectedRouteSymbol = SimpleLineSymbol(
+                  SimpleLineSymbol.Style.SOLID,
+                  Color.GREEN, 5f
+                )
+                Graphic(geometry, selectedRouteSymbol).also {
+                  graphicsOverlay.graphics.add(it)
+                }
+
+                // collapse the bottom sheet
+                bottomSheetBehavior.state = BottomSheetBehavior.STATE_COLLAPSED
+              }
+          }
+
+          // when the route is solved, hide the FAB
+          directionFab.visibility = View.GONE
+        } else {
+          val message = "Error loading route parameters."
+          Log.e(TAG, message)
+          Toast.makeText(this, message, Toast.LENGTH_LONG).show()
+        }
+      } catch (e: Exception) {
+        val message = "Error creating route parameters: ${e.message}"
+        Log.e(TAG, message)
+        Toast.makeText(this, message, Toast.LENGTH_LONG).show()
+      }
+    }
+  }
+
+  /**
+   * Set up the source, destination and route symbols
+   */
+  private fun setupSymbols() {
```

Comment:
I noticed that on my device the pin symbols look huge! They could do with a 50% reduction in size.

Reply:
Hopefully it's not too small now! I changed it to 30 width and height

Reply:
(was 40 I think)